examples/qml_minimal/src/cxxqt_object.rs

```diff
@@ -15,36 +15,37 @@ mod ffi {
     unsafe extern "C++" {
         include!("cxx-qt-lib/include/qt_types.h");
         type QString = cxx_qt_lib::QString;
+        type QPoint = cxx_qt_lib::QPoint;
     }
 
     // ANCHOR: book_data_struct
-    pub struct Data {
+    // TODO: remove book links
+    // ANCHOR_END: book_data_struct
+
+    // ANCHOR: book_rustobj_struct
+    #[cxx_qt::qobject]
+    pub struct MyObject {
+        #[qproperty]
         number: i32,
+        #[qproperty]
         string: UniquePtr<QString>,
```

Comment:
Having an opaque type C++ type stored inside a UniquePtr in the struct then causes the build to fail as `Opaque types are currently required to be [Sized](https://doc.rust-lang.org/std/marker/trait.Sized.html) and [Unpin](https://doc.rust-lang.org/std/marker/trait.Unpin.html).` https://cxx.rs/extern-rust.html#opaque-rust-types and the QString is opaque so uses PhantonPinned which doesn't implement Unpin ;-/

Reply:
Using `string: Box<UniquePtr<QString>>` looks like it might work :thinking:

Reply:
It does, but how do we express this to the developer. Or do they need to write `Box<UniquePtr<T>>` when they want to have T as a property type ? And how do we know that this type needs special conversions to T for C++ ? And what happens to the getters/setters on the Rust side? Do they return `Box<UniquePtr<T>>` ? etc

Then do we have a special type such as `Owned<T>` or something?

Reply:
Solved with https://github.com/dtolnay/cxx/pull/1091